# Lista de exercícios (I) - Pandas

### (1) - Importe a biblioteca Pandas para o projeto.

In [ ]:
import pandas as pd
import numpy as np

### (2) - Importe o conjunto de dados casa_para_venda.csv e armazene-o em um pd.DataFrame.

In [ ]:
df = pd.read_csv('/content/drive/MyDrive/casa_para_venda.csv')

### (3) - Inspecione o dataframe com o método print( ).

In [ ]:
print(df)

                 city  area  ...  fire insurance (R$)  total (R$)
0           São Paulo    70  ...                   42        5618
1           São Paulo   320  ...                   63        7973
2        Porto Alegre    80  ...                   41        3841
3        Porto Alegre    51  ...                   17        1421
4           São Paulo    25  ...                   11         836
...               ...   ...  ...                  ...         ...
10687    Porto Alegre    63  ...                   22        1926
10688       São Paulo   285  ...                  191       19260
10689  Rio de Janeiro    70  ...                   78        7390
10690  Rio de Janeiro   120  ...                  155       14020
10691       São Paulo    80  ...                   22        1587

[10692 rows x 13 columns]


### (4) - Inspecione as 10 primeiras linhas do dataframe.

In [ ]:
df.head(10)

city  area  ...  fire insurance (R$)  total (R$)
0       São Paulo    70  ...                   42        5618
1       São Paulo   320  ...                   63        7973
2    Porto Alegre    80  ...                   41        3841
3    Porto Alegre    51  ...                   17        1421
4       São Paulo    25  ...                   11         836
5       São Paulo   376  ...                  121        8955
6  Rio de Janeiro    72  ...                   25        2750
7       São Paulo   213  ...                   41        7253
8       São Paulo   152  ...                  191       16440
9  Rio de Janeiro    35  ...                   30        2955

[10 rows x 13 columns]

### (5) - Inspecione às 5 últimas linhas do dataframe.

In [ ]:
df.tail()

city  area  ...  fire insurance (R$)  total (R$)
10687    Porto Alegre    63  ...                   22        1926
10688       São Paulo   285  ...                  191       19260
10689  Rio de Janeiro    70  ...                   78        7390
10690  Rio de Janeiro   120  ...                  155       14020
10691       São Paulo    80  ...                   22        1587

[5 rows x 13 columns]

### (6) - Obtenha algumas metainformações do dataframe. Quais e quantas colunas? De quais tipos? Há dados faltantes?

In [ ]:
df.shape[1]

13

In [ ]:
df.dtypes

city                   object
area                    int64
rooms                   int64
bathroom                int64
parking spaces          int64
floor                  object
animal                 object
furniture              object
hoa (R$)                int64
rent amount (R$)        int64
property tax (R$)       int64
fire insurance (R$)     int64
total (R$)              int64
dtype: object

In [ ]:
df.isna().any()

city                   False
area                   False
rooms                  False
bathroom               False
parking spaces         False
floor                  False
animal                 False
furniture              False
hoa (R$)               False
rent amount (R$)       False
property tax (R$)      False
fire insurance (R$)    False
total (R$)             False
dtype: bool

### (7) - Obtenha algumas informações estatísticas resumidas no dataframe. Qual o valor mais caro do condomínio? E do aluguel?

In [ ]:
df.loc[df['total (R$)'].idxmax()]

city                   Belo Horizonte
area                              155
rooms                               1
bathroom                            4
parking spaces                      0
floor                               4
animal                      not acept
furniture               not furnished
hoa (R$)                      1117000
rent amount (R$)                 2790
property tax (R$)                  64
fire insurance (R$)                38
total (R$)                    1120000
Name: 255, dtype: object

### R - O valor total mais caro do condomínio é de 1 milhão e 120 mil reais, de uma casa da cidade de Belo Horizonte.

In [ ]:
df.loc[df['rent amount (R$)'].idxmax()]

city                       São Paulo
area                             700
rooms                              4
bathroom                           7
parking spaces                     8
floor                              -
animal                         acept
furniture              not furnished
hoa (R$)                           0
rent amount (R$)               45000
property tax (R$)               8750
fire insurance (R$)              677
total (R$)                     54430
Name: 2182, dtype: object

### R - O valor de aluguel mais caro do condomínio é de 45 mil reais de uma casa em São Paulo.

In [ ]:
df[['total (R$)', 'rent amount (R$)']].describe()

total (R$)  rent amount (R$)
count  1.069200e+04      10692.000000
mean   5.490487e+03       3896.247194
std    1.648473e+04       3408.545518
min    4.990000e+02        450.000000
25%    2.061750e+03       1530.000000
50%    3.581500e+03       2661.000000
75%    6.768000e+03       5000.000000
max    1.120000e+06      45000.000000

### (8) - Selecione a coluna 'city' com a notação de atributo (ponto)

In [ ]:
df.city

0             São Paulo
1             São Paulo
2          Porto Alegre
3          Porto Alegre
4             São Paulo
              ...      
10687      Porto Alegre
10688         São Paulo
10689    Rio de Janeiro
10690    Rio de Janeiro
10691         São Paulo
Name: city, Length: 10692, dtype: object

### (9) Qual o nome da cidade no index 35?

In [ ]:
df.loc[35, 'city']

'Porto Alegre'

### (10) Selecione a coluna 'rent amount'

In [ ]:
df['rent amount (R$)']

0         3300
1         4960
2         2800
3         1112
4          800
         ...  
10687     1478
10688    15000
10689     6000
10690    12000
10691     1400
Name: rent amount (R$), Length: 10692, dtype: int64

### (11) Selecione a cidade no índice 100 usando o operador iloc (todas as colunas). Qual o valor do aluguel?

In [ ]:
df.iloc[100, :]

city                    Porto Alegre
area                              50
rooms                              1
bathroom                           1
parking spaces                     1
floor                              1
animal                         acept
furniture              not furnished
hoa (R$)                         350
rent amount (R$)                 950
property tax (R$)                 50
fire insurance (R$)               14
total (R$)                      1364
Name: 100, dtype: object

### R - A cidade localizacional de tal casa é Porto Alegre e o valor do aluguel é exatamente 950 reais.

### (12) Selecione a cidade no índice 123 com o operador iloc (todas as colunas). Quantos quartos têm?

In [ ]:
df.iloc[123, :]

city                       São Paulo
area                              94
rooms                              3
bathroom                           2
parking spaces                     2
floor                              1
animal                         acept
furniture              not furnished
hoa (R$)                        1337
rent amount (R$)                3570
property tax (R$)                140
fire insurance (R$)               46
total (R$)                      5093
Name: 123, dtype: object

### R - A cidade localizacional de tal casa é São Paulo e há 3 quartos presentes nesta casa.

### (13) - Selecione às últimas 8 linhas e as colunas 'area', 'rooms', 'bathroom' e 'floor' usando o operador iloc. Qual é o menor valor do aluguel? 

In [ ]:
df.tail(8).iloc[:, [1, 2, 3, 5]]

area  rooms  bathroom floor
10684    98      2         1     1
10685    83      3         2    11
10686   150      3         3     8
10687    63      2         1     5
10688   285      4         4    17
10689    70      3         3     8
10690   120      2         2     8
10691    80      2         1     -

In [ ]:
df['rent amount (R$)'].min()

450

### R - O menor valor do aluguel é de 450 reais.

In [ ]:
df.tail(8)['rent amount (R$)'].min()

1400

### R - O menor valor do aluguel das últimas 8 linhas do dataframe é de 1400 reais.

### (14) - Importe o conjunto de dados casa_para_venda e defina a coluna city como index.

In [ ]:
df_1 = pd.read_csv('/content/drive/MyDrive/casa_para_venda.csv', index_col = 'city')

### (15) - Inspecione o dataframe.

In [ ]:
df_1.head()

area  rooms  ...  fire insurance (R$)  total (R$)
city                       ...                                 
São Paulo       70      2  ...                   42        5618
São Paulo      320      4  ...                   63        7973
Porto Alegre    80      1  ...                   41        3841
Porto Alegre    51      2  ...                   17        1421
São Paulo       25      1  ...                   11         836

[5 rows x 12 columns]

### (16) - Selecione a linha com o rótulo de index 'Campinas' (usando o operador loc). Qual foi a saída?

In [ ]:
df_1.loc['Campinas', :]

area  rooms  ...  fire insurance (R$)  total (R$)
city                   ...                                 
Campinas    46      1  ...                    8        1181
Campinas   330      4  ...                  121        9129
Campinas   208      3  ...                   48        3328
Campinas   250      3  ...                   22        4178
Campinas    48      1  ...                   21        2185
...        ...    ...  ...                  ...         ...
Campinas    92      3  ...                   22        2669
Campinas    83      2  ...                   47        4781
Campinas   140      1  ...                   66        7012
Campinas   150      3  ...                   53        3739
Campinas   250      1  ...                   34        2836

[853 rows x 12 columns]

### R - A saída foi de um conjunto de dados com 853 linhas e 12 colunas.

### (17) - Renomeie a coluna 'city' para 'cidade'.

In [ ]:
# Renomeação do índice do segundo dataframe:
df_1.rename_axis(index = {'city': 'cidade'}, inplace = True)

In [ ]:
# Renomeação da coluna do primeiro dataframe:
df.rename(columns = {'city': 'cidade'}, inplace = True)

### (18) - Verifique se todas as observações na coluna 'cidade' são iguais à São Paulo. Salve a resposta em uma variável.

In [ ]:
(df['cidade'] == 'São Paulo').all()

False

In [ ]:
sp = (df['cidade'] == 'São Paulo').all()

### (19) - Remova a coluna 'furniture' do conjunto de dados.

In [ ]:
df.drop(columns = 'furniture', inplace = True)

### (20) - Crie uma matriz de correlação para o DataFrame.

In [ ]:
df.corr()

area     rooms  ...  fire insurance (R$)  total (R$)
area                 1.000000  0.193796  ...             0.188078    0.051799
rooms                0.193796  1.000000  ...             0.565148    0.134597
bathroom             0.226766  0.733763  ...             0.676399    0.208339
parking spaces       0.193983  0.617510  ...             0.597348    0.148684
hoa (R$)             0.006890  0.007139  ...             0.029535    0.955024
rent amount (R$)     0.180742  0.541758  ...             0.987343    0.264490
property tax (R$)    0.039059  0.075252  ...             0.105661    0.218344
fire insurance (R$)  0.188078  0.565148  ...             1.000000    0.254911
total (R$)           0.051799  0.134597  ...             0.254911    1.000000

[9 rows x 9 columns]

## SEÇÃO DE EXERCÍCIOS FINALIZADA.